### PR TITLE
Add Solution to LC Problem - 2032

### LINKED ISSUE #3480: [Feature Request]:  Add Solution to LC Problem - 2032
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Problem Name - [Two Out of Three](https://leetcode.com/problems/two-out-of-three/)

I need to add solution to this Problem with detailed explanation

### Use Case

[Two Out of Three](https://leetcode.com/problems/two-out-of-three/)

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by sreevidya-16**
## Related Issue

Fixes #3480 Issue

## Description

[Two Out of Three](https://leetcode.com/problems/two-out-of-three/)

Given three integer arrays nums1, nums2, and nums3, return a distinct array containing all the values that are present in at least two out of the three arrays. You may return the values in any order.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+id: 2032-two-out-of-three
+title: Two Out of Three
+sidebar_label: 2032 - Two Out of Three
+tags: [Array, Hash Table, Set, C++]
+description: Return a distinct array containing all the values that are present in at least two out of the three arrays.
+sidebar_position: 1
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md:7]: replace `sidebar_position: 1` with `sidebar_position: 33`

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+Therefore, the overall time complexity is:
+
+\[ O(n1 + n2 + n3) \]
+
+The space complexity is dominated by the space required for the `unordered_map`, which stores up to O(n1 + n2 + n3) distinct elements.
+
+### Code
+
+### C++
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md:66]: ### Bad Practice

```md
  class Solution {
    public:
        vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
            unordered_map<int, char> mp;
            for (auto i : nums1) mp[i] = 'A';
            for (auto i : nums2) {
                if (mp[i] == 'A' || mp[i] == 'C') mp[i] = 'C';
                else mp[i] = 'B';
            }
            for (auto i : nums3) {
                if (mp[i] == 'C' || mp[i] == 'B' || mp[i] == 'A') mp[i] = 'C';
                else mp[i] = 'K';
            }
            for (auto i : nums1) {
                if (mp[i] == 'B' || mp[i] == 'K' || mp[i] == 'C') mp[i] = 'C';
                else mp[i] = 'A';
            }
            vector<int> ans;
            for (auto i : mp) if (i.second == 'C') ans.push_back(i.first);
            return ans;
        }
    };
```

### Good Practices

````md
```cpp
   class Solution {
    public:
        vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
            unordered_map<int, char> mp;
            for (auto i : nums1) mp[i] = 'A';
            for (auto i : nums2) {
                if (mp[i] == 'A' || mp[i] == 'C') mp[i] = 'C';
                else mp[i] = 'B';
            }
            for (auto i : nums3) {
                if (mp[i] == 'C' || mp[i] == 'B' || mp[i] == 'A') mp[i] = 'C';
                else mp[i] = 'K';
            }
            for (auto i : nums1) {
                if (mp[i] == 'B' || mp[i] == 'K' || mp[i] == 'C') mp[i] = 'C';
                else mp[i] = 'A';
            }
            vector<int> ans;
            for (auto i : mp) if (i.second == 'C') ans.push_back(i.first);
            return ans;
        }
    };
```
````

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                else mp[i] = 'A';
+            }
+            vector<int> ans;
+            for (auto i : mp) if (i.second == 'C') ans.push_back(i.first);
+            return ans;
+        }
+    };
+
+### JavaScript
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md:91]: ### bad practices

```md
    /**
     * @param {number[]} nums1
     * @param {number[]} nums2
     * @param {number[]} nums3
     * @return {number[]}
     */
    var twoOutOfThree = function(nums1, nums2, nums3) {

        const res = []
        const arr = new Set ([...nums1,...nums2,...nums3])

        arr.forEach((val)=>{
            if(nums1.includes(val) && nums2.includes(val) || nums1.includes(val) 
            && nums3.includes(val) || nums2.includes(val) && nums3.includes(val)){
                res.push(val)
            }
        })
        
        return res    
    };
```

### Good Practices

````mdx
```js
    /**
     * @param {number[]} nums1
     * @param {number[]} nums2
     * @param {number[]} nums3
     * @return {number[]}
     */
    var twoOutOfThree = function(nums1, nums2, nums3) {

        const res = []
        const arr = new Set ([...nums1,...nums2,...nums3])

        arr.forEach((val)=>{
            if(nums1.includes(val) && nums2.includes(val) || nums1.includes(val) 
            && nums3.includes(val) || nums2.includes(val) && nums3.includes(val)){
                res.push(val)
            }
        })
        
        return res    
    };
```
````

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                res.push(val)
+            }
+        })
+        
+        return res    
+    };
+
+### Java
+
+    class Solution {
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md:115]: ### Bad Practices

```md
    class Solution {
        public List<Integer> twoOutOfThree(int[] nums1, int[] nums2, int[] nums3) {
            Set<Integer> set = new HashSet<>();
            for(int num : nums1)
                set.add(num);
            
            List<Integer> result = new ArrayList<>();
            Set<Integer> set2 = new HashSet<>();
            for(int num : nums2){
                if(set.contains(num) && !set2.contains(num))
                    result.add(num);
             
                set2.add(num);
            }
            
            for(int num : nums3)
                if(!result.contains(num) && (set.contains(num) || set2.contains(num)))
                    result.add(num);
            
            return result;
        }
    }
```

### Good Practices

````mdx
```java
    class Solution {
        public List<Integer> twoOutOfThree(int[] nums1, int[] nums2, int[] nums3) {
            Set<Integer> set = new HashSet<>();
            for(int num : nums1)
                set.add(num);
            
            List<Integer> result = new ArrayList<>();
            Set<Integer> set2 = new HashSet<>();
            for(int num : nums2){
                if(set.contains(num) && !set2.contains(num))
                    result.add(num);
             
                set2.add(num);
            }
            
            for(int num : nums3)
                if(!result.contains(num) && (set.contains(num) || set2.contains(num)))
                    result.add(num);
            
            return result;
        }
    }
```
````

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@sreevidya-16, Thanks for the code contribution! I made some suggestions and guidelines to follow in the comments. Please update your code and let me know when you’re finished so I can review it again. 

[Guidelines for Writing DSA Problem Solutions](https://github.com/orgs/CodeHarborHub/discussions/3369)

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
+---
+id: 2032-two-out-of-three
```
[the remark this answers is not in the document]
Comment (sreevidya-16) [reply dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md]: Done

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md (hunk 1 above)
Comment (sreevidya-16) [reply dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md]: Done

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md (hunk 3 above)
Comment (sreevidya-16) [reply dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md]: Done

File: dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md (hunk 4 above)
Comment (sreevidya-16) [reply dsa-solutions/lc-solutions/2000-2099/2032-Two-Out-of-three-problem.md:117]: Done

**Comment by sreevidya-16** [comment]
I have done all the changes you suggested to me 
Kindly view them and let me know if there are any mistakes

**Comment by Ajay-Dhangar** [comment]
@sreevidya-16, Please compare your code with our format

````md
---
id: enter-id-for-routes
title: Title of Problem
sidebar_label: Number of Problem
tags: [tag1, tag2, ...., tagn]
description: Write description for SEO in few words and in only string and can't use any HTML tags in this line
sidebar_position: enter position number
---

## Problem Statement 

### Problem Description

<!-- Content here -->

### Examples

<!-- Problem Examples here -->

### Constraints

<!-- Problem Constraints here -->

## Solution of Given Problem

### Intuition and Approach

<!-- Content here -->

### Approaches

<!-- Content here -->

#### Codes in Different Languages

<Tabs>
  <TabItem value="JavaScript" label="JavaScript" default>
  <SolutionAuthor name="@username"/>
   ```javascript
    // code here
    ```

  </TabItem>
  <TabItem value="TypeScript" label="TypeScript">
  <SolutionAuthor name="@username"/> 
   ```typescript
    // code here
    ```

  </TabItem>
  <TabItem value="Python" label="Python"> 
  <SolutionAuthor name="@username"/>
   ```python
    // code here
    ```

  </TabItem>
  <TabItem value="Java" label="Java">
  <SolutionAuthor name="@username"/>
   ```java
    // code here
    ```
  </TabItem>
  <TabItem value="cpp" label="C++">
  <SolutionAuthor name="@username"/>
   ```cpp
    // code here
    ```
  </TabItem>  
</Tabs>

### Complexity Analysis

<!-- Content here -->

## Video Explanation of Given Problem

<Tabs>
  <TabItem value="en" label="English">
    <Tabs>
      <TabItem value="javascript" label="JavaScript">
        <LiteYouTubeEmbed
          id="video_id"
          params="autoplay=1&autohide=1&showinfo=0&rel=0"
          title="Problem Explanation | Solution | Approach"
          poster="maxresdefault"
          webp 
        />
      </TabItem>
      <TabItem value="python" label="Python">
        <LiteYouTubeEmbed
          id="video_id"
          params="autoplay=1&autohide=1&showinfo=0&rel=0"
          title="Problem Explanation | Solution | Approach"
          poster="maxresdefault"
          webp 
        />
      </TabItem>
      <TabItem value="java" label="Java">
        <LiteYouTubeEmbed
          id="video_id"
          params="autoplay=1&autohide=1&showinfo=0&rel=0"
          title="Problem Explanation | Solution | Approach"
          poster="maxresdefault"
          webp 
        />
      </TabItem>
    </Tabs>
  </TabItem>

  <TabItem value="hi" label="Hindi">
    <LiteYouTubeEmbed
      id="video_id"
      params="autoplay=1&autohide=1&showinfo=0&rel=0"
      title="Problem Explanation | Solution | Approach"
      poster="maxresdefault"
      webp 
    />
  </TabItem>
</Tabs>

---

<h2>Authors:</h2>

<div style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', gap: '10px'}}>
{['github_username'].map(username => (
 <Author key={username} username={username} />
))}
</div>
```` 

and resolve your issue what's wrong in your code.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**